src/coreComponents/linearAlgebra/interfaces/petsc/PetscVector.cpp

```diff
@@ -214,17 +214,26 @@ void PetscVector::axpby( real64 const alpha,
   }
 }
 
-void PetscVector::pointwiseProduct( PetscVector const & x,
-                                    PetscVector & y ) const
+void PetscVector::pointwiseProduct( PetscVector const & x )
 {
   GEOS_LAI_ASSERT( ready() );
   GEOS_LAI_ASSERT( x.ready() );
-  GEOS_LAI_ASSERT( y.ready() );
+  GEOS_LAI_ASSERT_EQ( localSize(), x.localSize() );
   GEOS_LAI_ASSERT_EQ( globalSize(), x.globalSize() );
```

Comment:
Checking for `localSize` should be enough. If equal, same `globalSize` is automatically implied

Reply:
Fixed